Gordon360/ApiControllers/HousingController.cs

```diff
@@ -61,10 +64,16 @@ public IHttpActionResult SaveApplication([FromBody] ApartmentAppViewModel apartm
                 }
                 throw new BadInputException() { ExceptionMessage = errors };
             }
+            //get token data from context, username is the username of current logged in person
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+
+            string modifierId = _accountService.GetAccountByUsername(username).GordonID;
 
-            int apartAppId = apartmentAppDetails.AprtAppID; // Set the apartAppId to -1 to indicate that an application ID was not passed by the frontend
-            string modifierId = _accountService.GetAccountByUsername(apartmentAppDetails.Username).GordonID;
             string sessionId = Helpers.GetCurrentSession().SessionCode;
+
+            int apartAppId = apartmentAppDetails.AprtAppID; // The apartAppId is set to -1 if an existing application ID was not yet known by the frontend
+            // string modifierId = _accountService.GetAccountByUsername(apartmentAppDetails.Username).GordonID;
```

Comment:
Is this line here for the sake of testing? I see that you defined the modifierId a few lines above. Did you leave this here because you haven't decided which one to use?

Reply:
The one commented out is the old method, where it gets the id from the one passed in the view model, whereas the new method gets the user's info from the auth data of the current user. I left the old line commented out until we have a chance to test the new idea. It should be removed only after we have successfully tested this new version